rave-compiler/src/main/java/com/uber/rave/compiler/RaveWriter.java

```diff
@@ -156,7 +157,7 @@ private MethodSpec generateConstructor(@NonNull RaveIR raveIR) {
         builder.endControlFlow();
         List<MethodSpec> concreteSpecs = new ArrayList<>(raveIR.getNumClasses());
         for (ClassIR classIR : raveIR.getClassIRs()) {
-            MethodSpec specificSpec = generateConcreteMethodSpec(classIR);
+            MethodSpec specificSpec = generateConcreteMethodSpec(raveIR, classIR);
```

Comment:
Why are we passing in RaveIR here? Isn't the mode the only thing we need? This method shouldn't require the raveIR.